src/yunohost/certificate.py

```diff
@@ -639,6 +639,9 @@ def _prepare_certificate_signing_request(domain, key_file, output_folder):
     # Set the domain
     csr.get_subject().CN = domain
 
+    # Include xmpp-upload subdomain as subject alternate names
+    csr.add_extensions([crypto.X509Extension("subjectAltName", False, "DNS:xmpp-upload." + domain)])
```

Comment:
To answer your previous post, and if I understood correctly, naively I would just have something here that checks that there's a good expected DNS record for xmpp-upload. In pseudo-code:

```text
if [DNS record for wildcard exist] OR ([DNS record for xmpp-upload.domain.tld exists] AND [the value of that DNS record match the value for @])
then
    add xmpp-upload to the csr extensions
fi
```

Reply:
Actually, I was reluctant to perform those extra DNS queries each time a certificate renewal attempt is made. But maybe I was too thrifty on this. After all, renewal attempt only occur once a day.
Let's do that, and possibly improve this later.

Reply:
hmyea if you want there's also a possibility to interface with the diagnosis system from 3.8 for this

Reply:
But nvm about that. Renewal attempts do indeed only occur once a day, and *actual* renewal (which is what we are concerned about here) is supposedly only happening like every 2.5 months or so